processor/src/main/java/org/mapstruct/ap/processor/MethodRetrievalProcessor.java

```diff
@@ -80,19 +93,60 @@ public int getPriority() {
         return 1;
     }
 
+    private List<SourceMethod> retrievePrototypeMethods(TypeMirror typeMirror, MapperConfig mapperConfig) {
+        if ( typeMirror == null || typeMirror.getKind() == TypeKind.VOID ) {
+            return Collections.emptyList();
+        }
+
+        TypeElement typeElement = asTypeElement( typeMirror );
+        List<SourceMethod> methods = new ArrayList<SourceMethod>();
+        for ( ExecutableElement executable : getAllEnclosedExecutableElements( elementUtils, typeElement ) ) {
+
+            ExecutableType methodType = typeFactory.getMethodType( typeElement, executable );
+            List<Parameter> parameters = typeFactory.getParameters( methodType, executable );
+            boolean containsTargetTypeParameter = SourceMethod.containsTargetTypeParameter( parameters );
+
+            // prototype methods don't have prototypes themselves
```

Comment:
So to make sure we're all on the same page:

* The default setting is EXPLICIT
* If AUTO_INHERIT_FROM_CONFIG is enabled (either on the mapper itself or the config class), configurations from methods declared in the config class are automatically inherited to methods on the mapper with assignable parameter/return types
* If more than one method from the config class are suitable candidates to inherit mappings from, an ambiguity error is raised for the concerned mapping method(s)
* `@InheritConfiguration` / `@InheritInverseConfiguration` needs to be given explicitly to inherit config from other methods on the same mapper or its super-types

Is this what we all agree on? If so, is there anything left to decide on?

One minor issue I see is that one needs to declare a base mapping method which actually should be usable by mapper clients and whose configuration should be auto-inherited by other methods, twice: once on the config class (to make it auto-inheritable) and another time on the mapper (to make it invocable). I can live with that, I wouldn't expect such method to be intended as usable in the most cases anyways.